tests/st/calicoctl/test_node_run.py

```diff
@@ -0,0 +1,44 @@
+# Copyright (c) 2015-2017 Tigera, Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from tests.st.test_base import TestBase
+from tests.st.utils.docker_host import DockerHost
+from tests.st.utils.exceptions import CommandExecError
+from tests.st.utils.utils import retry_until_success
+import os
+
+"""
+Test calicoctl node run
+
+The `calicoctl node run` command is tested fairly throughouly by other tests.  Here we
+want to test passing env vars through `calicoctl node run` into the container.
+"""
+
+
+class TestNodeRun(TestBase):
+    
+    def test_node_run_with_env_vars(self):
+        """
+        Set some CALICO_ and FELIX_ env vars and see that they make it through to the
+        Node container
+        """
+        os.environ['CALICO_TESTVAL'] = 'dummyVal'
+        os.environ['FELIX_TESTVAL'] = 'felixDummyVal'
+
+        with DockerHost('host', dind=False) as host:
+            calico_out = host.execute('printenv CALICO_TESTVAL')
+            felix_out = host.execute('printenv FELIX_TESTVAL')
+
+            self.assertEquals(calico_out, 'dummyVal')
+            self.assertEquals(felix_out, 'felixDummyVal')
```

Comment:
You can tell the Dockerhost to start calico-node inside it by simply passing it the argument `start_calico=True` when you create it.